src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -266,7 +266,13 @@ private void RunAssembly(string assemblyName, TestFilter filter)
                     }
                 }
                 else
-                    TestLog.Info("NUnit failed to load " + assemblyName);
+                {
+                    var msgNode = loadResult.SelectSingleNode("properties/property[@name='_SKIPREASON']");
+                    if (msgNode != null && (new[] { "contains no tests", "Has no TestFixtures" }).Any(msgNode.GetAttribute("value").Contains))
+                        TestLog.Info("Assembly " + assemblyName + " contains no tests supported by this runner: " + _activeRunner.GetType().AssemblyQualifiedName);
```

Comment:
It might not be absolutely correct, but I think the message should be directed toward the general user, and inform about possibly user mistakes. If there is a NUnitAdapter/NUnit mismatch issue that is possibly a bug in the adapter, as any adapter version should ideally be able to run any NUnit version (and I know it is not really true this will happen).   But I am not sure we need to surface that in this message.  Alternatively, replace "this" with the explicit name and version we're running.  "contains no NUnit3 tests that can be run by NUnitAdapter version xxx"